packages/yew/src/functional/hooks/mod.rs

```diff
@@ -1,19 +1,52 @@
 mod use_context;
 mod use_effect;
+mod use_memo;
 mod use_reducer;
 mod use_ref;
 mod use_state;
 
 pub use use_context::*;
 pub use use_effect::*;
+pub use use_memo::*;
 pub use use_reducer::*;
 pub use use_ref::*;
 pub use use_state::*;
 
-use crate::functional::{HookUpdater, CURRENT_HOOK};
-use std::cell::RefCell;
-use std::ops::DerefMut;
-use std::rc::Rc;
+use crate::functional::{hook, AnyScope, HookContext, HookUpdater};
+
+/// A trait that is implemented on hooks.
+///
+/// A hook is usually defined via `#[hooks]`. Please refer to its documentation on how to implement
+/// hooks.
```

Comment:
```suggestion
/// A hook is usually defined via [`#[hook]`][hook]. Please refer to its documentation on how to implement
/// hooks.
```
Can we also add the guarantees that `[hook]` ensures here?